modules/nlu/src/backend/index.ts

```diff
@@ -1,34 +1,42 @@
 import 'bluebird-global'
 import * as sdk from 'botpress/sdk'
 
+import { Config } from '../config'
+
 import api from './api'
-import FTWrapper from './fasttext/fasttext.wrapper'
-import ScopedNlu from './scopednlu'
-import { initBot, initModule } from './setup'
+import { registerMiddleware } from './middleware'
+import fastText from './tools/fastText'
 
-export type SDK = typeof sdk
+import ScopedEngine from './engine'
+import { DucklingEntityExtractor } from './pipelines/entities/duckling_extractor'
+import Storage from './storage'
 
-const botScopedNlu: Map<string, ScopedNlu> = new Map<string, ScopedNlu>()
+const nluByBot: EngineByBot = {}
 
-const onServerStarted = async (bp: SDK) => {
-  await initModule(bp, botScopedNlu)
+const onServerStarted = async (bp: typeof sdk) => {
+  Storage.ghostProvider = botId => bp.ghost.forBot(botId)
 
-  const config = await bp.config.getModuleConfig('nlu')
-  if (config.fastTextPath) {
-    FTWrapper.changeBinPath(config.fastTextPath)
-  }
+  const globalConfig = (await bp.config.getModuleConfig('nlu')) as Config
+  globalConfig.fastTextPath && fastText.configure(globalConfig.fastTextPath)
+  DucklingEntityExtractor.configure(globalConfig.ducklingEnabled, globalConfig.ducklingURL)
```

Comment:
As a matter of consistency, we might want to separate the interaction with duckling in a tool just like we did with FT. then handled the configuration logic with this tool, what do you think ? Maybe we could do this only when we integrate the offline duckling though.

Reply:
I'd wait for the offline duckling as you suggest or until we break the DRY principle